data_registry/process_manager/task/coverage.py

```diff
@@ -0,0 +1,64 @@
+import gzip
+import shutil
+import tempfile
+from pathlib import Path
+
+import ocdscardinal
+
+from data_registry import models
+from data_registry.process_manager.util import TaskManager, skip_if_not_started
+from exporter.util import Export
+
+
+def get_keys_for_subschema(coverage, subschema):
+    return (key for key in coverage if key.endswith(subschema))
+
+
+class Coverage(TaskManager):
+    final_output = True
+
+    def get_export(self):
+        return Export(self.job.pk, "coverage")
+
+    def run(self):
+        export = self.get_export()
+        export.lock()
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tmpdir = Path(tmpdirname)
+            infile = tmpdir / "tmp.jsonl"
+            with gzip.open(Export(self.job.pk, "full.jsonl.gz").path) as i, infile.open("wb") as o:
+                shutil.copyfileobj(i, o)
+            coverage = ocdscardinal.coverage(str(infile))
+            mapping = {
+                "tenders_count": ("/tender/",),
+                "tenderers_count": ("/tender/tenderers[]",),
+                "tenders_items_count": ("/tender/items[]",),
+                "parties_count": ("/parties[]",),
+                "awards_count": ("/awards[]",),
+                "awards_items_count": ("/awards[]/items[]",),
+                "awards_suppliers_count": ("/awards[]/suppliers[]",),
+                "contracts_count": ("/contracts[]",),
+                "contracts_items_count": ("/contracts[]/items[]",),
+                "contracts_transactions_count": ("/contracts[]/transactions[]",),
+                "documents_count": get_keys_for_subschema(coverage, "documents[]"),
+                "plannings_count": ("/planning/",),
+                "milestones_count": get_keys_for_subschema(coverage, "milestones[]"),
+                "amendments_count": get_keys_for_subschema(coverage, "amendments[]"),
+            }
+            for attr, paths in mapping.items():
+                setattr(self.job, attr, sum(coverage.get(path, 0) for path in paths))
+            self.job.context["coverage"] = coverage
+            self.job.save(update_fields=["modified", "coverage", *mapping])
```

Comment:
I just realized that "coverage" here should have been "context" – but we are switching to a separate field anyway.